File: FinApp/Data/Entities/Currency.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Text;

namespace Data.Entities
{
    public class Currency
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal ExchangeRate { get; set; }

        public ICollection<Account> Accounts { get; set; }

        public Currency()
        {
            Accounts = new Collection<Account>();
        }
    }
}

File: FinApp/FinApp/ClientApp/src/app/models/currency.ts
export class Currency {
  Id: number;
  Name: string;
  ExchengeRate: number;

  constructor(id: number,
    name: string,
    exchengeRate: number) {
    this.Id = id;
    this.Name = name;
    this.ExchengeRate = exchengeRate;
  }
}
